internal/controller/numaflowcontroller/numaflowcontroller_controller.go

```diff
@@ -241,20 +241,37 @@ func (r *NumaflowControllerReconciler) reconcile(
 		return ctrl.Result{}, err
 	}
 
+	// Generate the manifest for the NumaflowController based on the version specified in the controller spec. Which is used for
+	// creating the Deployment and other resources.
 	newVersion := controller.Spec.Version
 	newVersionTargetObjs, err := determineTargetObjects(controller, newVersion, namespace)
 	if err != nil {
 		return ctrl.Result{}, fmt.Errorf("unable to determine the target objects for the new version %s: %w", newVersion, err)
 	}
-
 	numaLogger.Debugf("found %d target objects associated with NumaflowController version %s", len(newVersionTargetObjs), newVersion)
 
-	// apply controller - this handles syncing in the cases in which our Controller  isn't updating
+	// Generate the manifest for the old version of the NumaflowController, which is used for calculating the diff between the old and new versions.
+	oldVersion := controller.Status.LastAppliedVersion
+
+	// TODO: This is temporary logic to handle the case where the controller is being created for the first time.
+	// Will remove this code once we have LastAppliedVersion set in the controller.
+	if oldVersion == "" {
+		numaLogger.Debug("No previous version found, assuming this is the first reconciliation")
+		oldVersion = newVersion // if no previous version, assume it's the same as the new version
+	}
+
+	oldVersionTargetObjs, err := determineTargetObjects(controller, oldVersion, namespace)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("unable to determine the target objects for the old version %s: %w", oldVersion, err)
```

Comment:
What if for some reason the ConfigMap no longer exists? In that case, how about we log a Warning but just move forward anyway?